File: server-pedidos-java/src/main/java/br/edu/utfpr/serverpedidosjava/repository/CategoriaRepository.java
package br.edu.utfpr.serverpedidosjava.repository;

import br.edu.utfpr.serverpedidosjava.entity.Categoria;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface CategoriaRepository extends JpaRepository<Categoria, Integer> {
    List<Categoria> findByNomeContainingIgnoreCase(String nome);

    Page<Categoria> findByNomeContainingIgnoreCase(String nome, Pageable pageable);
}

File: server-pedidos/src/main/kotlin/br/edu/utfpr/serverpedidos/repository/CategoriaRepository.kt
package br.edu.utfpr.serverpedidos.repository

import br.edu.utfpr.serverpedidos.entity.Categoria
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface CategoriaRepository : JpaRepository<Categoria, Int> {
    fun findByNomeContainsIgnoreCase(nome: String): List<Categoria>
    fun findByNomeContainsIgnoreCase(nome: String, pageable: Pageable): Page<Categoria>
}
